Sources/Public/CalendarView.swift

```diff
@@ -162,6 +162,14 @@ public final class CalendarView: UIView {
     setNeedsLayout()
   }
 
+  public override func hitTest(_ point: CGPoint, with event: UIEvent?) -> UIView? {
+    // If the user interacts with the calendar, we should clear out any existing
+    // `scrollToItemContext` that might be leftover from the initial layout process.
+    scrollToItemContext = nil
```

Comment:
I used to do this in `scrollViewDidScroll`, but that case didn't cover the drag gesture auto scroll that this PR adds. The real logic is "whenever the user interacts with the calendar, set this to nil" - I think `hitTest` is the most reliable place? `touchesBegan` doesn't get called if a different view or gesture captures the touch.

Reply:
hmmm, I think this function having side effects isn't ideal, maybe is risky.

> leftover from the initial layout process.

 can you elaborate? Hopefully we can find a better place to put this

Reply:
We use a `scrollToItemContext` to represent a programmatic scroll to a certain item / position (with or without animation).

When the calendar's bounds changes, we set a non-animated `scrollToItemContext` to scroll to the same offset that we were at prior to the bounds change. This enables us to maintain our scroll position on bounds changes (rotation, initial layout as the calendar receives its initial/final size).

For as long as there's a non-nil `scrollToItemContext`, the calendar won't scroll like normal. That's why the previous logic would set it to nil in `scrollViewDidScroll`, effectively enabling the calendar to scroll normally. We need to do this same logic for the auto-scroll behavior that this PR adds.

I could just duplicate the logic to nil out the `scrollToItemContext` in the auto-scroll code. I thought maybe I could solve it a little bit earlier by detecting any kind of user interaction.